### PR TITLE
Add method to get agreements by term

### PR DESCRIPTION
**Pull request by hatched**
With the `terms.getAgreementsByTerms` method you can now pass it a list of terms and it'll return the terms which need to be agreed to still.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

- Connect to an environment where you have agreed to terms, get those terms by running `app.getAgreements((e, t) => console.log(t))`
- Then taking one of those agreements run `app.getAgreementsByTerms([<term-name>], (e, t) => console.log(t))` and it should return null, or the term if you haven't yet agreed to it. On jujugui.org you can use the `hatch/test-term1` term if you need one you haven't yet agreed to.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6769/

File: jujugui/static/gui/src/app/jujulib/terms.js [hunk 1]
```diff
@@ -161,6 +164,42 @@ var module = module;
       };
       var url = this.url + '/agreements';
       return jujulib._makeRequest(this.bakery, url, 'GET', null, handler);
+    },
+
+    /**
+      Calls to check if the supplied terms have been agreed to. If they have
+      it returns an empty array, else it'll return the terms which need to be
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/jujulib/terms.js]: I might be mistaken, but doesn't line 171 say this will return null?

**Comment by huwshimi** [comment]
Thanks @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**